src/Compilers/Core/MSBuildTaskTests/GenerateMSBuildAnalyzerConfigTests.cs

```diff
@@ -0,0 +1,303 @@
+            TaskItem item2 = new TaskItem("c:\\file2.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "def456" } });
+            TaskItem item3 = new TaskItem("c:\\file3.cs", new Dictionary<string, string> { { "ItemType", "AdditionalFiles" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "ghi789" } });
+            TaskItem item4 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "AdditionalFiles" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "jkl012" } });
+
+            TaskItem property1 = new TaskItem("Property1", new Dictionary<string, string> { { "Value", "abc123" } });
+            TaskItem property2 = new TaskItem("Property2", new Dictionary<string, string> { { "Value", "def456" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2, item3, item4 },
+                PropertyItems = new[] { property1, property2 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+msbuild_property.Property1 = abc123
+msbuild_property.Property2 = def456
+
+[c:/file1.cs]
+msbuild_item.Compile.ToRetrieve = abc123
+msbuild_item.AdditionalFiles.ToRetrieve = jkl012
+
+[c:/file2.cs]
+msbuild_item.Compile.ToRetrieve = def456
+
+[c:/file3.cs]
+msbuild_item.AdditionalFiles.ToRetrieve = ghi789
+", result);
+        }
+
+        [Fact]
+        public void ItemIsNotFullyQualifiedPath()
+        {
+            TaskItem item1 = new TaskItem("file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+            TaskItem item2 = new TaskItem("..\\file2.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+            TaskItem item3 = new TaskItem("someDir\\otherDir\\thirdDir\\..\\file3.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2, item3 }
+            };
+            configTask.Execute();
+
+            string executingLocation = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal($@"is_global = true
+
+[{Path.GetFullPath("file1.cs").Replace('\\', '/')}]
+msbuild_item.Compile.ToRetrieve = abc123
+
+[{Path.GetFullPath("..\\file2.cs").Replace('\\', '/')}]
+msbuild_item.Compile.ToRetrieve = abc123
+
+[{Path.GetFullPath("someDir\\otherDir\\thirdDir\\..\\file3.cs").Replace('\\', '/')}]
```

Comment:
How is this test functioning? I can't see where these files are written to disk hence can't see how this not throwing.

Reply:
It's not writing anything to disk. MSBuild takes the value of the `ItemSpec` and appends the Working Directory to it when you ask for the full path. The tests here are just doing that same calculation so it works without having to hard code absolute paths.

I'll update the test to make it a little clearer what's going on.


---
In reply to: [419849389](https://github.com/dotnet/roslyn/pull/43617#discussion_r419849389) [](ancestors = 419849389)